#! /bin/sh /usr/share/dpatch/dpatch-run
## h323_no_exit.dpatch by Tzafrir Cohen <tzafrir.cohen@xorcom.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Prevent the need to run 'make' twice to build h323.

@DPATCH@
diff -urNad asterisk-1.4.1~dfsg~/channels/Makefile asterisk-1.4.1~dfsg/channels/Makefile
--- asterisk-1.4.1~dfsg~/channels/Makefile	2007-03-17 23:50:15.000000000 +0200
+++ asterisk-1.4.1~dfsg/channels/Makefile	2007-03-17 23:51:50.000000000 +0200
@@ -36,10 +36,6 @@
   H323LIB=-lh323_NetBSD_x86_r
 endif
 
-ifeq ($(wildcard h323/libchanh323.a),)
-  CC_MODS:=$(filter-out chan_h323,$(CC_MODS))
-endif
-
 ifndef OPENH323DIR
   OPENH323DIR=$(HOME)/openh323
 endif
@@ -62,23 +58,10 @@
 clean::
 	rm -f busy.h ringtone.h gentone
 	$(MAKE) -C misdn clean
-
-ifneq ($(wildcard h323/Makefile.ast),)
-  include h323/Makefile.ast
-H323LDFLAGS+=-Wl,--version-script=h323/noexport.map
-clean::
 	$(MAKE) -C h323 clean
-else
-h323/libchanh323.a h323/Makefile.ast:
-	$(CMD_PREFIX) $(MAKE) -C h323
-	$(CMD_PREFIX) rm -f ../main/asterisk
-	$(CMD_PREFIX) echo "***************************************************************"
-	$(CMD_PREFIX) echo
-	$(CMD_PREFIX) echo "********** Re-run 'make' to pick up H.323 parameters **********"
-	$(CMD_PREFIX) echo
-	$(CMD_PREFIX) echo "***************************************************************"
-	$(CMD_PREFIX) exit 1
-endif
+
+h323/libchanh323.a:
+	$(CMD_PREFIX) $(MAKE) -C $(@D) $(@F)
 
 gentone: gentone.c
 	$(ECHO_PREFIX) echo "   [LD] $^ -> $@"
@@ -98,7 +81,7 @@
 chan_alsa.o: busy.h ringtone.h
 
 ifeq ($(OSARCH),linux-gnu)
-chan_h323.so: chan_h323.o h323/libchanh323.a h323/Makefile.ast
+chan_h323.so: chan_h323.o h323/libchanh323.a
 	$(ECHO_PREFIX) echo "   [LD] $^ -> $@"
 	$(CMD_PREFIX) $(CXX) $(PTHREAD_CFLAGS) $(ASTLDFLAGS) $(SOLINK) $(H323LDFLAGS) -o $@ $< h323/libchanh323.a $(H323LDLIBS)
 else
